src/plugins/formulas/formulas.js

```diff
@@ -197,64 +265,198 @@ export class Formulas extends BasePlugin {
       });
     }
 
+    // HyperFormula events:
+    this.engine.on('valuesUpdated', (...args) => this.onHFvaluesUpdated(...args));
+
     super.enablePlugin();
   }
 
   /**
    * Disables the plugin functionality for this Handsontable instance.
    */
   disablePlugin() {
-    // TODO add tests for this line
-    this.hyperformula.destroy();
-
     super.disablePlugin();
   }
 
+  /**
+   * Triggered on `updateSettings`.
+   */
+  updatePlugin() {
+    const hotSettings = this.hot.getSettings();
+
+    this.#settings = hotSettings[PLUGIN_KEY];
+
+    if (this.#settings.engine) {
+      this.engine.updateConfig(mergeEngineSettings(this.#settings));
+    }
+
+    if (isDefined(this.#settings.sheetName) && this.#settings.sheetName !== this.sheetName) {
+      this.switchSheet(this.#settings.sheetName);
+    }
+
+    super.updatePlugin();
+  }
+
   /**
    * Destroys the plugin instance.
    */
   destroy() {
+    unregisterEngine(this.engine, this.hot.guid);
+
     super.destroy();
   }
 
-  onAfterLoadData() {
-    if (!this.enabled) {
+  /**
+   * Add a sheet to the shared HyperFormula instance.
+   *
+   * @param {string} [sheetName] The new sheet name. If not provided, will be auto-generated by HyperFormula.
+   * @param {Array} sheetData Data passed to the shared HyperFormula instance. Has to be declared as an array of
+   * arrays - array of objects is not supported in this scenario.
+   * @returns {boolean} `false` if the data format is unusable, `true` otherwise.
+   */
+  addSheet(sheetName, sheetData) {
+    if (!isAoA(sheetData)) {
+      error('The provided data should be an array of arrays.');
+
+      return false;
+    }
+
+    if (this.engine.doesSheetExist(sheetName)) {
+      error('Sheet with the provided name already exists.');
+
+      return false;
+    }
+
+    const actualSheetName = this.engine.addSheet(sheetName ?? void 0);
+
+    if (this.engine.isItPossibleToReplaceSheetContent(actualSheetName, sheetData)) {
+      this.engine.setSheetContent(actualSheetName, sheetData);
+    }
+
+    return true;
```

Comment:
No message if the data set is not used? And false `true` returned because our isAoA accepted the format but HF `isItPossibleToReplaceSheetContent` did not